src/primitives.jl

```diff
@@ -160,25 +160,19 @@ Return a vector containing only the non-null entries of `X`,
 unwrapping `Nullable` entries. A copy is always returned, even when
 `X` does not contain any null values.
 """
-function dropnull(X::AbstractVector)                 # -> AbstractVector
-    if !(Nullable <: eltype(X))
+function dropnull{T}(X::AbstractVector{T})
+    if !(Nullable <: T) && !(T <: Nullable)
         return copy(X)
+    else
+        Y = filter(x->!_isnull(x), X)
+        res = similar(Y, eltype(T))
```

Comment:
It's interesting that `eltype(Any) == Any`, I didn't know that. Else that code wouldn't have worked. There's a method definition in Base specially for that, so I guess we can rely on it.